dag_in_context/src/optimizations/body_contains.rs

```diff
@@ -11,7 +11,7 @@ fn captured_expr_rule_for_ctor(ctor: Constructor) -> Option<String> {
     let pat = ctor.construct(|field| field.var());
     let actions = ctor.filter_map_fields(|field| {
         (field.purpose == Purpose::CapturedExpr)
-            .then(|| format!("(BodyContainsExpr {pat} {e})", e = field.var()))
+            .then(|| format!("(BodyContainsExpr {e} {e})", e = field.var()))
```

Comment:
BodyContainsExpr now correctly stores the body (root of the region) instead of the parent Loop or If